projects/kit/components/input-phone-international/input-phone-international.component.ts

```diff
@@ -0,0 +1,246 @@
+    },
+    viewProviders: [
+        tuiDropdownOptionsProvider({
+            limitWidth: 'fixed',
+            align: 'right',
+        }),
+    ],
+})
+export class TuiInputPhoneInternationalComponent
+    extends AbstractTuiControl<string>
+    implements TuiFocusableElementAccessor
+{
+    /* eslint-disable @typescript-eslint/member-ordering */
+    protected readonly options = inject(TUI_INPUT_PHONE_INTERNATIONAL_OPTIONS);
+    protected readonly countryIsoCode$ = new BehaviorSubject(this.options.countryIsoCode);
+    protected readonly mask$ = new BehaviorSubject<MaskitoOptions | null>(null);
+    protected readonly countriesNames$ = inject(TUI_COUNTRIES);
+    protected readonly textfieldOptions = inject(TuiTextfieldOptionsDirective, {
+        optional: true,
+    });
+
+    protected open = false;
+    protected textfieldValue = '';
+
+    @ViewChild(MaskitoDirective, {read: ElementRef})
+    private readonly inputPhone?: ElementRef<HTMLInputElement>;
+
+    @ViewChild(TuiSelectDirective, {read: ElementRef})
+    private readonly countrySelect?: ElementRef<HTMLSelectElement>;
+
+    @Input()
+    public countries = this.options.countries;
+
+    @Output()
+    public readonly countryIsoCodeChange = this.countryIsoCode$.pipe(
+        distinctUntilChanged(),
+        skip(1),
+    );
+
+    @Input('countryIsoCode')
+    public set isoCode(code: TuiCountryIsoCode) {
+        this.countryIsoCode$.next(code);
+    }
+
+    public get nativeFocusableElement(): HTMLElement | null {
+        return this.inputPhone && !this.computedDisabled
+            ? this.inputPhone.nativeElement
+            : null;
+    }
+
+    public get focused(): boolean {
+        return Boolean(
+            tuiIsNativeFocused(this.countrySelect?.nativeElement) ||
+                tuiIsNativeFocused(this.inputPhone?.nativeElement) ||
+                this.open,
+        );
+    }
+
+    public onPaste(event: Event, phonesMetadata: MetadataJson | null): void {
+        if (
+            !tuiIsInputEvent(event) ||
+            !phonesMetadata ||
+            (!event.inputType.includes('Drop') && !event.inputType.includes('Paste'))
+        ) {
+            return;
+        }
+
+        const newValue = event.data || '';
+        const prefixedValue = newValue.startsWith(CHAR_PLUS)
+            ? newValue
+            : CHAR_PLUS + newValue;
+
+        if (validatePhoneNumberLength(prefixedValue) === 'TOO_SHORT') {
+            return;
+        }
+
+        const countryIsoCode = maskitoGetCountryFromNumber(prefixedValue, phonesMetadata);
+
+        if (countryIsoCode) {
+            this.countryIsoCode$.next(countryIsoCode);
+        }
+    }
+
+    public onItemClick(isoCode: TuiCountryIsoCode): void {
+        this.open = false;
+        this.countryIsoCode$.next(isoCode);
+        this.nativeFocusableElement?.focus();
+    }
+
+    constructor() {
+        super();
+        combineLatest([this.countryIsoCode$, this.options.metadata])
```

Comment:
This looks like it can be optimized through signals. Are we sure there's no double load of metadata with 2 subscriptions? You can keep it like that for now, maybe I'll take a stab at it later when I will be moving directives to the host and refactoring it to use new signal control.

Reply:
I'll rewrite code (using signals) in next PR